bids/analysis/variables.py

```diff
@@ -183,6 +183,8 @@ def load_variables(layouts, default_duration=0, entities=None, columns=None,
 
     for name, grp in _df.groupby(['factor', 'condition']):
         data = grp.apply(pd.to_numeric, errors='ignore')
+        # Subjects are often labeled numerically; ignore this
+        data['subject'] = grp['subject']
```

Comment:
To grab files with zero-padded subject labels, you need this. There may be other entities we need to do this with...